sympy/stats/rv_interface.py

```diff
@@ -339,6 +340,61 @@ def factorial_moment(X, n, condition=None, **kwargs):
     """
     return expectation(FallingFactorial(X, n), condition=condition, **kwargs)
 
+def median(X, evaluate=True, **kwargs):
+    r"""
+    Calculuates the median of the probability distribution.
+    Mathematically, median of Probability distribution is defined as all those
+    values of `m` for which the following condition is satisfied
+
+    .. math::
+        P(X\geq m)\geq 1/2 \hspace{5} \text{and} \hspace{5} P(X\leq m)\geq 1/2
+
+    Parameters
+    ==========
+
+    X: The random expression whose median is to be calculated.
+
+    Returns
+    =======
+
+    The FiniteSet or an Interval which contains the median of the
+    random expression.
```

Comment:
We need to be clear whether this returns the set of all such values

Reply:
How about returning the centre of the interval containing the possible median values.

Reply:
That seems reasonable. Have you ever heard of that being used as a definition for median? It would be good to have a name for this definition or a reference.

Reply:
That seems reasonable. Have you ever heard of that being used as a definition for median? It would be good to have a name for this definition or a reference.

Reply:
Here the `Interval` completely is an `median`. It means that any real value in the `Interval` can be said as the `median` or the `Interval` as a whole.

Reply:
The reference to the` median` definition is [this link](https://en.wikipedia.org/wiki/Median#Probability_distributions). It also has the example of how the `median` of a distribution can be an `Interval`.

Reply:
> We need to be clear whether this returns the set of all such values

It returns `FiniteSet` in case when `median` is finite (Ex. as in Die) and return `Interval` in case we have an Interval of real values which satisfy the condition of `median`.
Die should not return an Interval as it never takes any value between 3 and 4, So the median is finite and equal to `{3, 4}`